qutebrowser/browser/urlmarks.py

```diff
@@ -244,17 +264,35 @@ class BookmarkManager(UrlMarkManager):
         - removed gets emitted with the URL as argument.
     """
 
+    def _bookmarks_directory(self):
+        """Return the directory containing the bookmarks.
+
+        For bookmarks are data, return by default a folder inside the standard
+        data directory. In order not to break the compatibility with an older
+        behavior, if bookmarks are found in the standard configuration
+        directory, return this directory instead.
+        """
+        config_dir = standarddir.config()
+        if config_dir is not None:
+            inconfigdir = os.path.join(standarddir.config(), 'bookmarks')
+        else:
+            inconfigdir = None
+        indatadir = os.path.join(standarddir.data(), 'bookmarks')
+        if inconfigdir is not None and os.path.isdir(inconfigdir):
+            return inconfigdir
+        else:
+            return indatadir
```

Comment:
Hi,
Thanks for the comment.
Jakub Klinkovský <notifications@github.com> writes:

[...]

> On second thought, if `standarddir.config()` can return `None`, are we sure that `standarddir.data()` doesn't?

The constructor of urlmark already checks the standarddir.data. May be
we can recheck here for sake of clarity or defer the check to the
constructor.

-- 
Konubinix
GPG Key    : 7439106A
Fingerprint: 5993 BE7A DA65 E2D9 06CE  5C36 75D2 3CED 7439 106A